zerver/tornado/event_queue.py

```diff
@@ -1126,6 +1112,10 @@ def process_message_update_event(
 
     for user_data in users:
         user_profile_id = user_data["id"]
+        # If `user_id` isn't present, then this must be a message embeds event, and
+        # we don't need to send notifications, so short-circuit.
+        message_sender_id = event_template.get("user_id", user_profile_id)
```

Comment:
Are sure the `user_id` field in that structure the sender?  

Also, how does this logic short-circuit?  

I think it'd be reasonable for an `if 'user_id' not in event_template` to not call `maybe_enqueue_notifications_for_message_update` at all, by reordering this function a bit.

Reply:
OK, I've confirmed that `user_id` in this object, if present, is the ID of the user object that triggered the edit -- so not the sender of the message in all cases (topic edits can be done by other users).  And of course it'd be a bug if we later added an anyone-editable message type.

I think with our current featureset, only senders can edit their own messages' content, and I don't think any of our alerting features trigger off content in the topic (E.g. you can't put a mention there).  Though it does strike me when saying this that moving messages between streams (to one where you have configured notifications for all messages) may not be a code path we've tested properly.  Let's leave that digression for follow-up work, since I think it's a rabbit hole.

Having now read the code, I think we can write this for this commit:
```
if "user_id" in event_template:
    # This is inaccurate: the user we'll get here will be the sender if the message's
    # content was edited, which is typically where we might send new notifications.  However, 
    # for topic/stream edits, it could be another user.  We may need to adjust the format
    # for update_message events to address this issue.
    message_sender_id = event_template["user_id"]
else:
    # This is also inaccurate, but usefully so.  Events without a `user_id` field come from the 
    # do_update_embedded_data code path, and represent just rendering previews; there
    # should be no real content changes.  This is a hack to simplify notifications logic for this
    # code path.  TODO: Change this to short-circuit more directly.
    message_sender_id = user_profile_id
```

Reply:
> I think with our current featureset, only senders can edit their own messages' content, and I don't think any of our alerting features trigger off content in the topic

Yeah, that was the reasoning. I should have given that context.

Reply:
> Though it does strike me when saying this that moving messages between streams (to one where you have configured notifications for all messages) may not be a code path we've tested properly.

I think the message editor should never receive notifications (even if the editor and sender are different people, and even if we in the future allowed others to edit message content), because they are the cause. This is in the same spirit as someone not receiving notifications for messages they themselves sent, or for mentioning themselves.

Reply:
> Also, how does this logic short-circuit?

Hmm, OK "short-circuit" is probably a wrong term to use here. What I meant was, it doesn't really matter what we're calling `sender_id`, because we know the message isn't doing to be notifiable anyways, since its an embed event. I'll put a `assert not is_notifiable` there so that that's clear.

Reply:
Added those comments.

Reply:
Ahh, I see, that's a good point.  I wonder if we can address this somewhat by changing the `sender_id` parameter to be called `acting_user`?